### PR TITLE
Move errors to Event Log Viewer, remove duplicates

### PR DESCRIPTION
**Pull request by brucearmstrong**
- Instead of a Snackbar, display errors in the Event Log Viewer.
- Remove duplicate error messages.
- Overriding the value of generateDmeHostAddress() with always-working EU DME. Code comment about why this is done.

### CODE REVIEW

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java [hunk 1]
```diff
@@ -274,7 +273,13 @@ private boolean registerClient() throws ExecutionException, InterruptedException
             Log.e(TAG, "Could not generate host");
             host = "wifi.dme.mobiledgex.net";   //fallback host
         }
-//        host = "eu-mexdemo.dme.mobiledgex.net";
+
+        // The generateDmeHostAddress() call above will generate a hostname that includes the
+        // MCC+MNC of your current provider. Example: AT&T is 310-410.dme.mobiledgex.net.
+        // Since most of these values haven't been mapped to a live DME, here we override
+        // the host with a known good value for a DME that has our app instance deployed.
+        host = "eu-mexdemo.dme.mobiledgex.net";
```
Comment (ah1053) [inline android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:281]: 👍

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java [hunk 2]
```diff
@@ -609,7 +620,9 @@ protected void onPostExecute(Boolean cloudletFound) {
             if (!cloudletFound) {
                 findCloudletStatusText = "Failed to find cloudlet. " + findCloudletStatusText;
                 Log.e(TAG, findCloudletStatusText);
-                showErrorMsg(findCloudletStatusText);
+                if (!errorShown) {
+                    showErrorMsg(findCloudletStatusText);
+                }
```
Comment (ah1053) [inline android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:625]: If I understood correctly this check will remove duplicate error messages, I am just concerned if the order errors appears matters. (Or if there is a datetime string next to the error) 

Also maybe QoSPosBackgroundRequest() needs the same check if it is still active in the new app version.

Comment (brucearmstrong) [reply android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:625]: There is a timestamp.

Good point about QoSPosBackgroundRequest. PQOS/PQOE isn't currently supported by any carrier, and this code should probably be disabled.

**Review by ah1053 — APPROVED** [review]

File: android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java (hunk 2 above)
Comment (brucearmstrong) [reply android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java:625]: @ah1053 Now I see that it is actually working currently, I guess using random simulator data. So I'll leave the code intact. I've added the same duplicate check to the QOS code.

**Pull request merged**